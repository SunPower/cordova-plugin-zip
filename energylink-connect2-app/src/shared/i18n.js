import { useMemo } from 'react'
import { useSelector } from 'react-redux'

import * as languages from './locales'
import * as localizedComponents from './localizedComponents'

const getTranslate = translations => (key, ...params) => {
  let value = translations[key] || key
  if (params) {
    params.forEach((p, idx) => {
      value = value.replace(`{${idx}}`, p)
    })
  }
  return value
}

export const useI18n = () => {
  const language = useSelector(state => state.language)
  const { locale = 'en' } = language
  return useMemo(() => getTranslate(languages[locale]), [locale])
}

export const useI18nComponent = () => {
  const language = useSelector(state => state.language)
  const { locale = 'en' } = language || {}

  return useMemo(
    () => componentName => {
      const path = `${locale}_${componentName}`
      const Component = localizedComponents[path] || {}
      if (typeof Component !== 'function') {
        throw new Error(`Component not found: ${path}`)
      }
      return Component()
    },
    [locale]
  )
}
