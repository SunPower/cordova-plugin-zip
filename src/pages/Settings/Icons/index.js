export { ReactComponent as WifiIcon } from './wifi-icon.svg'
export { ReactComponent as BatteryIcon } from './battery-icon.svg'
