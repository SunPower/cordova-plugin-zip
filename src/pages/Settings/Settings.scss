.menu-modal {
  padding-left: 1rem !important;
  padding-right: 1rem !important;
}
