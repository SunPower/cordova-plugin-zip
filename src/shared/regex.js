export const PASSWORD_REGEXP = /^(?=.*[a-z])(?=.*\d)[A-Za-z\d]{8,}$/
export const EMAIL_REGEXP = /^\S+@\S+\.\S+$/
export const PHONE_REGEXP = /^(1\s?)?((([0-9]{3}))|[0-9]{3})[\s-]?[0-9]{3}[\s-]?[0-9]{4}$/
