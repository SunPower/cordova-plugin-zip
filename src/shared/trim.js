export const trimObject = obj => {
  return Object.entries(obj)
    .map(([k, v]) => ({
      [k]: typeof v === 'string' || v instanceof String ? v.trim() : v
    }))
    .reduce((acc, kv) => ({ ...acc, ...kv }), {})
}
