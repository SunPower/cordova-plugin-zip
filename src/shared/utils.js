export const either = (condition, whenTrue, whenFalse) =>
  condition ? whenTrue : whenFalse
