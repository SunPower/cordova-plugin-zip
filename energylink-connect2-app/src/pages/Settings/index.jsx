import { compose, isNil, path, prop } from 'ramda'
import React from 'react'
import { useSelector, useDispatch } from 'react-redux'
import { useHistory } from 'react-router'

import Toggler from 'components/Toggler'
import paths from 'routes/paths'
import { useI18n } from 'shared/i18n'
import { either } from 'shared/utils'
import { LOGOUT } from 'state/actions/auth'
import { SET_SHOW_CHECKLIST } from 'state/actions/settings'
import { MENU_HIDE } from 'state/actions/ui'

function Settings() {
  const t = useI18n()
  const dispatch = useDispatch()
  const history = useHistory()

  const { showPrecommissioningChecklist } = useSelector(prop('global'))
  const name = useSelector(path(['user', 'data', 'name']))

  const logout = () => {
    dispatch(LOGOUT())
    dispatch(MENU_HIDE())
    history.push(paths.PROTECTED.ROOT.path)
  }

  return (
    <main className="full-height pl-15 pr-15">
      <h5 className="is-uppercase is-block is-full-width has-text-centered is-bold mb-30">
        {t('SETTINGS')}
      </h5>
      <section>
        <h6 className="has-text-white">{t('APP_SETTINGS')}</h6>

        <article>
          <div className="mt-20">
            <Toggler
              text={t('SETTING_CHECKLIST')}
              checked={showPrecommissioningChecklist}
              onChange={compose(dispatch, SET_SHOW_CHECKLIST)}
            />
          </div>
        </article>

        <article role="button" onClick={logout}>
          <hr className=" mb-10" />
          <h6 className="has-text-white mb-10">{t('BTN_LOG_OUT')}</h6>
          {either(!isNil(name), <p>{t('LOGGED_IN', name)}</p>)}
          <hr className="mt-10" />
        </article>
      </section>
    </main>
  )
}

export default Settings
