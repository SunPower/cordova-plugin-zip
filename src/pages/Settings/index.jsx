import React from 'react'
import clsx from 'clsx'
import { useSelector } from 'react-redux'
import ModalLayout from '../../components/ModalLayout'
import SocialFooter from '../../components/SocialFooter'
import BlockGrid from '../../components/BlockGrid'
import BlockItem from '../../components/BlockItem'
import SunPowerImage from '../../components/SunPowerImage'
import paths from '../Router/paths'
import { useI18n } from '../../shared/i18n'
import { WifiIcon, BatteryIcon } from './Icons'

import './Settings.scss'

const wifiState = {
  CONNECTED: {
    text: 'CONNECTED',
    state: 'success'
  },
  UNCONFIGURED: {
    text: 'NOT_CONNECTED',
    state: 'error'
  },
  CHECKING: {
    text: 'CHECKING_CONNECTION',
    state: 'warn'
  }
}

function Settings({ history, location, className }) {
  const t = useI18n()
  const classes = clsx('menu-modal', className)
  const wifiStateEnum = useSelector(
    state =>
      state.wifi &&
      state.wifi.collectorStatus &&
      state.wifi.collectorStatus.WifiStatRptCtnt &&
      state.wifi.collectorStatus.WifiStatRptCtnt.wifiStatEnum
  )
  const wifi = wifiState[wifiStateEnum] || wifiState.CHECKING
  return (
    <ModalLayout
      className={classes}
      history={history}
      imageHeader={<SunPowerImage inverse />}
      from={location && location.state && location.state.from}
      hasBackButton
    >
      <BlockGrid location={location}>
        <BlockItem
          className="full-width-title"
          title={t('WIFI_SETTINGS')}
          to={paths.MENU_WIFI_SETTINGS}
          icon={<WifiIcon />}
          label={t(wifi.text)}
          labelState={wifi.state}
        />
        <BlockItem
          title={t('BATTERY_SETTINGS')}
          to={paths.MENU_BATTERY_SETTINGS}
          icon={<BatteryIcon />}
        />
      </BlockGrid>

      <div className="mt-50 mb-20 pl-10 pr-10">
        <SocialFooter />
      </div>
    </ModalLayout>
  )
}

export default Settings
