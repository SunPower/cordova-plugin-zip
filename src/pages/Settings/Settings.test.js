import React from 'react'
import * as reactRedux from 'react-redux'
import { shallow } from 'enzyme'
import Settings from '.'
import * as i18n from '../../shared/i18n'

describe('Settings page', () => {
  beforeEach(() => {
    jest
      .spyOn(i18n, 'useI18n')
      .mockImplementation(path => (key, ...params) =>
        `${key.toUpperCase()} ${params.join('_')}`.trim()
      )
  })
  it('renders without crashing', () => {
    jest.spyOn(reactRedux, 'useSelector').mockImplementation(() => 'CONNECTED')
    const component = shallow(<Settings />)
    expect(component).toMatchSnapshot()
  })
})
