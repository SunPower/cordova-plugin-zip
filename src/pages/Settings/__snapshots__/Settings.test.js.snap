// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings page renders without crashing 1`] = `
<ModalLayout
  className="menu-modal"
  hasBackButton={true}
  imageHeader={
    <SunPowerImage
      inverse={true}
    />
  }
>
  <BlockGrid>
    <BlockItem
      className="full-width-title"
      icon={<ForwardRef(SvgWifiIcon) />}
      label="CONNECTED"
      labelState="success"
      title="WIFI_SETTINGS"
      to="/menu/wifi-settings"
    />
    <BlockItem
      icon={<ForwardRef(SvgBatteryIcon) />}
      title="BATTERY_SETTINGS"
      to="/menu/battery-settings"
    />
  </BlockGrid>
  <div
    className="mt-50 mb-20 pl-10 pr-10"
  >
    <SocialFooter />
  </div>
</ModalLayout>
`;
